src/gsym/types.rs

```diff
@@ -0,0 +1,41 @@
+pub const GSYM_MAGIC: u32 = 0x4753594d;
+pub const GSYM_VERSION: u16 = 1;
+
+pub struct Header {
+    pub magic: u32,
+    pub version: u16,
+    pub addr_off_size: u8,
+    pub uuid_size: u8,
+    pub base_address: u64,
+    pub num_addrs: u32,
+    pub strtab_offset: u32,
+    pub strtab_size: u32,
+    pub uuid: [u8; 20],
+}
+
+#[allow(dead_code)]
+pub struct FileInfo {
+    pub directory: u32,
+    pub filename: u32,
+}
+
+pub struct AddressInfo<'a> {
+    pub size: u32,
+    pub name: u32,
+    /// The raw data comprises a list of [`AddressData`].
+    pub data: &'a [u8],
+}
+
+pub struct AddressData<'a> {
+    /// The data type. Its value should be one of InfoType*.
+    pub typ: u32,
```

Comment:
I'd rather we express that connection by using a proper `enum` ala https://doc.rust-lang.org/reference/items/enumerations.html#custom-discriminant-values-for-fieldless-enumerations